src/engine/BMGame.php

```diff
@@ -939,8 +821,10 @@ protected function roll_active_dice_needing_values() {
 
     protected function update_game_state_specify_dice() {
         if (!$this->isWaitingOnAnyAction()) {
-            $this->prevSwingValueArrayArray = NULL;
-            $this->prevOptValueArrayArray = NULL;
+            foreach ($this->playerArray as $player) {
+                $player->prevSwingValueArray = array();
+                $player->prevOptValueArray = array();
+            }
```

Comment:
Yeah, this is in keeping with the change to initialisation philosophy, where now, all the arrays accessed via the convenience accessors should always exist. That gets around the problem that we had right at the beginning that we always had to check if the input was an array before doing a foreach.